spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/eureka/EurekaInstanceAutoConfiguration.java

```diff
@@ -114,20 +124,33 @@ private SanitizingEurekaInstanceConfigBean getDefaults() {
 		SanitizingEurekaInstanceConfigBean eurekaInstanceConfigBean = new SanitizingEurekaInstanceConfigBean(new InetUtils(inetUtilsProperties));
 		eurekaInstanceConfigBean.setHostname(hostname);
 		eurekaInstanceConfigBean.setIpAddress(ip);
-		eurekaInstanceConfigBean.getMetadataMap().put(INSTANCE_ID, instanceId);
+		Map<String, String> metadataMap = eurekaInstanceConfigBean.getMetadataMap();
+		metadataMap.put(INSTANCE_ID, instanceId);
+		metadataMap.put(ZONE, zoneFromUri(zoneUri));
 
 		return eurekaInstanceConfigBean;
 	}
 
+	private static String zoneFromUri(String defaultZoneUri) {
+		String hostname = null;
+		try {
+			hostname = new URI(defaultZoneUri).getHost();
```

Comment:
URI can throw a NullPullException if defaultZoneUri is null. Will that case never happen?